controllers/awsmachine_controller.go

```diff
@@ -418,30 +417,29 @@ func (r *AWSMachineReconciler) reconcileDelete(machineScope *scope.MachineScope,
 // findInstance queries the EC2 apis and retrieves the instance if it exists.
 // If providerID is empty, finds instance by tags and if it cannot be found, returns empty instance with nil error.
 // If providerID is set, either finds the instance by ID or returns error.
-func (r *AWSMachineReconciler) findInstance(scope *scope.MachineScope, ec2svc services.EC2Interface) (*infrav1.Instance, error) {
+func (r *AWSMachineReconciler) findInstance(machineScope *scope.MachineScope, ec2svc services.EC2Interface) (*infrav1.Instance, error) {
 	var instance *infrav1.Instance
 
 	// Parse the ProviderID.
-	//nolint:staticcheck
 	// Usage of noderefutil pkg would be removed in a future release.
-	pid, err := noderefutil.NewProviderID(scope.GetProviderID())
+	pid, err := scope.NewProviderID(machineScope.GetProviderID())
 	if err != nil {
 		//nolint:staticcheck
 		// Usage of noderefutil pkg would be removed in a future release.
-		if !errors.Is(err, noderefutil.ErrEmptyProviderID) {
+		if !errors.Is(err, scope.ErrEmptyProviderID) {
```

Comment:
We should leave this using **nodrefutil**.
(applies to the other instances)

Reply:
I am changing it because we upgraded and noderefutil has removed it

Reply:
Ok that makes sense.